cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -521,7 +524,8 @@ private void createLocalDatasets() throws IOException, DatasetManagementExceptio
           @Override
           public Void call() throws Exception {
             datasetFramework.addInstance(instanceSpec.getTypeName(), instanceId,
-                                         addLocalDatasetProperty(instanceSpec.getProperties()));
+                                         addLocalDatasetProperty(instanceSpec.getProperties()),
+                                         ownerAdmin.getOwner(program.getId().getParent()));
```

Comment:
I don't think this works in distributed mode. The `OwnerAdmin` talks to HBase directly, which needs CDAP token, which a program container won't have. Please check with @rohitsinha54 and @Nishith